mem_db.go

```diff
@@ -15,40 +40,42 @@ func init() {
 var _ DB = (*MemDB)(nil)
 
 type MemDB struct {
-	mtx sync.Mutex
-	db  map[string][]byte
+	mtx   sync.RWMutex
+	btree *btree.BTree
 }
 
 func NewMemDB() *MemDB {
 	database := &MemDB{
-		db: make(map[string][]byte),
+		btree: btree.New(32),
```

Comment:
what 32 is here?

Reply:
The "degree" of the B-tree, i.e. the number of items per tree node. Benchmarks showed 32 to be the sweet spot.

Reply:
can we add a const degree = 32 and a comment saying 32 is best according to benchmarks?

Reply:
Yep, already done